docs/my_documentation.rst

```diff
@@ -0,0 +1,365 @@
+Flask-Multipass
+===============
+
+Installation
+------------
+
+Installing Flask-Multipass is very easy using pip::
+
+    $ pip install Flask-Multipass
+
+Initialization
+--------------
+
+To start with Multipass, create your Flask application, load the prefered configuration and create the ``Multipass`` object by passing it the application:
+
+.. code-block:: python
+
+    from flask import Flask
+    from flask_multipass import Multipass
+
+    app = Flask(__name__)
+    app.config.from_pyfile('example.cfg')
+    multipass = Multipass(app)
+
+When you have more then one application or you are using an application factory you can also use ``init_app`` :
+
+.. code-block:: python
+
+    apps = Flask('test'), Flask('test')
+    multipass = Multipass()
+    for app in apps:
+        multipass.init_app(app)
+
+Configuration
+--------------
+
+The following configuration values exist for Flask-Multipass:
+
+====================================== =========================================
+``MULTIPASS_AUTH_PROVIDERS``           Dictionary of authentication providers
+``MULTIPASS_IDENTITY_PROVIDERS``       Dictionary of identification providers
+``MULTIPASS_PROVIDER_MAP``             Mapping of authentication providers to identification providers
+``MULTIPASS_IDENTITY_INFO_KEYS``       Keys used for identification
+``MULTIPASS_LOGIN_SELECTOR_TEMPLATE``  Template with selection of login providers
+``MULTIPASS_LOGIN_FORM_TEMPLATE``      Template with login form
+``MULTIPASS_LOGIN_ENDPOINT``           Endpoint linking to login page
+``MULTIPASS_LOGIN_URLS``               List of login URLs 
+``MULTIPASS_SUCCESS_ENDPOINT``         Endpoint linking to default page after successful login
+``MULTIPASS_FAILURE_MESSAGE``          Message to show after unsuccessfull login
+``MULTIPASS_FAILURE_CATEGORY``         Category of message when flashing after unsuccessful login
+``MULTIPASS_ALL_MATCHING_IDENTITIES``  If true, all matching identities are passed after successful authentication
+``MULTIPASS_REQUIRE_IDENTITY``         If true, ``IdentityRetrievalFailed`` is raised when no matching identities are found, othwerwise empty list is passed
+====================================== =========================================
+
+Configuration example can be found here
+
+< link to config.example.rst>
+
+Registering providers
+---------------------
+
+You can register new provider using the following commands. Remember that your provider class must be a subclass of either ``AuthProvider`` or ``IdentityProvider``:
+
+.. code-block:: python
+
+	class FooProvider(AuthProvider):
+		pass
+
+Then you pass this class to ``register_provider`` with second argument beeing the type name of the provider specified in configuration. Remember that type names must be unique. :
+
+.. code-block:: python
+
+	app = Flask('test')
+	app.config['MULTIPASS_AUTH_PROVIDERS'] = {
+		'test_auth_provider': {'type': 'test'}
+	}
+	multipass = Multipass(app)
```

Comment:
You've already written these lines above, there's no need to repeat them, as people should be following along.